src/db/state/record.rs

```diff
@@ -106,6 +108,32 @@ impl<'conn> State<'conn, RecordRow> {
         self.add_refs_impl(std::iter::once(alias), CitationKeyInsertMode::FailIfExists)
     }
 
+    /// Ensure that the given alias exists for this row.
+    ///
+    /// If the alias already exists and points to a different row, the canonical id of the other row is returned.
+    #[inline]
+    pub fn ensure_alias(&self, alias: &Alias) -> Result<Option<RemoteId>, rusqlite::Error> {
+        debug!(
+            "Ensuring alias '{alias}' refers to row_id '{}'",
+            self.row_id()
+        );
+        match get_row_id(&self.tx, alias)? {
+            Some(existing_row_id) => {
+                if existing_row_id == self.row_id() {
+                    return Ok(None);
+                } else {
+                    let RowData { canonical, .. } = get_row_data(&self.tx, existing_row_id)?;
```

Comment:
It bothers me somewhat that the function `get_row_data` is implemented twice, but I can see why it is done this way here. I might be tempted to manually inline the `get_row_data` call here.